geopandas/geodataframe.py

```diff
@@ -409,10 +409,18 @@ def _constructor(self):
         return GeoDataFrame
 
     def __finalize__(self, other, method=None, **kwargs):
-        """ propagate metadata from other to self """
-        # NOTE: backported from pandas master (upcoming v0.13)
-        for name in self._metadata:
-            object.__setattr__(self, name, getattr(other, name, None))
+        """propagate metadata from other to self """
+        # merge operation: using metadata of the left object
+        if method == 'merge':
+            for name in self._metadata:
```

Comment:
in theory you an inspect ``other.how``; if its ``right`` then you could us that. Should should also be cognizant of conflicting meta-data.